flashcard-quiz/src/main.rs

```diff
@@ -0,0 +1,246 @@
+use std::{
+    fs::File,
+    io::{stdin, stdout, Write as _},
+    path::Path,
+};
+
+use clap::Parser as _;
+use rand::seq::SliceRandom as _;
+
+/// Flashcard CLI - Create and manage flashcards for studying.
+#[derive(Debug, clap::Parser)]
+#[clap(version = env!("CARGO_PKG_VERSION"))]
+struct Args {
+    #[clap(subcommand)]
+    subcommand: Subcommand,
+}
+
+#[derive(Debug, clap::Subcommand)]
+enum Subcommand {
+    /// Add a new flashcard
+    Add { question: String, answer: String },
+    /// Edit an existing flashcard
+    Edit {
+        index: usize,
+        question: Option<String>,
+        answer: Option<String>,
+    },
+    /// Remove an existing flashcard
+    Remove { index: usize },
+    /// Show all flashcards
+    Show,
+    /// Start the quiz
+    Quiz {
+        #[clap(short, long, default_value = "5")]
+        num_questions: usize,
+    },
+}
+
+#[derive(Default, serde::Serialize, serde::Deserialize)]
+struct Quiz {
+    cards: Vec<Card>,
+}
+
+#[derive(serde::Serialize, serde::Deserialize)]
+struct Card {
+    question: String,
+    answer: String,
+}
+
+fn main() -> anyhow::Result<()> {
+    let args = Args::parse();
+
+    let mut quiz = load_or_create_quiz()?;
+
+    match args.subcommand {
+        Subcommand::Add { question, answer } => add_card(&mut quiz, &question, &answer),
+        Subcommand::Edit {
+            index,
+            question,
+            answer,
+        } => edit_card(&mut quiz, index - 1, question.as_deref(), answer.as_deref())?,
+        Subcommand::Remove { index } => remove_card(&mut quiz, index - 1)?,
+        Subcommand::Show => show_cards(&quiz),
+        Subcommand::Quiz { num_questions } => start_quiz(&quiz, num_questions)?,
+    }
+
+    save_quiz(&quiz)?;
+
+    Ok(())
+}
+
+fn load_or_create_quiz() -> anyhow::Result<Quiz> {
+    let path = Path::new("quiz.json");
+    if path.exists() {
+        let file = File::open(path)?;
+        let quiz = serde_json::from_reader(file)?;
+        Ok(quiz)
+    } else {
+        let quiz = Quiz::default();
+        Ok(quiz)
+    }
+}
+
+fn save_quiz(quiz: &Quiz) -> anyhow::Result<()> {
+    let file = File::create("quiz.json")?;
+    serde_json::to_writer(file, quiz)?;
+    Ok(())
+}
+
+fn add_card(quiz: &mut Quiz, question: &str, answer: &str) {
+    quiz.cards.push(Card {
+        question: question.to_string(),
+        answer: answer.to_string(),
+    });
+
+    println!(
+        "Added flashcard #{} with question: '{}' and answer: '{}'",
+        quiz.cards.len(),
+        question,
+        answer
+    );
+}
+
+fn edit_card(
+    quiz: &mut Quiz,
+    index: usize,
+    question: Option<&str>,
+    answer: Option<&str>,
+) -> anyhow::Result<()> {
+    let card = quiz
+        .cards
+        .get_mut(index)
```

Comment:
<details> 
  <summary>Corey's Second Bug Guess</summary>
Same index adjusting should happen here I think!

We would expect the User to type in the index's they see, so this needs to be `index - 1`
</details>